presto-hive/src/main/java/com/facebook/presto/hive/HivePartitionResult.java

```diff
@@ -59,9 +64,27 @@ public HivePartitionResult(
         return partitionColumns;
     }
 
-    public List<HivePartition> getPartitions()
+    public Iterator<HivePartition> getPartitions()
     {
-        return partitions;
+        return partitions.iterator();
+    }
+
+    public List<HivePartition> getPartitionsAsList(int maxSize)
+    {
+        ImmutableList.Builder<HivePartition> partitionList = ImmutableList.builder();
+        int listSize = 0;
+        Iterator<HivePartition> iterator = partitions.iterator();
+        while (iterator.hasNext()) {
+            HivePartition partition = iterator.next();
+            partitionList.add(partition);
+            if (listSize++ == maxSize) {
```

Comment:
We avoid using `++` or `--` inside an expression. Make that a separate statement.